src/txn/txn_rollback_to_stable.c

```diff
@@ -0,0 +1,394 @@
+/*-
+ * Copyright (c) 2014-2017 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#ifdef HAVE_TIMESTAMPS
+
+/*
+ * __txn_rollback_to_stable_check --
+ *	Ensure the rollback request is reasonable.
+ */
+static int
+__txn_rollback_to_stable_check(
+    WT_SESSION_IMPL *session, wt_timestamp_t *rollback_timestamp)
+{
+	WT_TXN_GLOBAL *txn_global = &S2C(session)->txn_global;
+	WT_TXN_STATE *s;
+	uint32_t i, session_cnt;
+
+	if (__wt_timestamp_cmp(
```

Comment:
I believe you need to acquire the `txn_global->rwlock` read lock for this function.  Also a comment here at the beginning of the code outlining what is reasonable.